SU2_CFD/include/numerics/NEMO/CNEMONumerics.hpp

```diff
@@ -155,20 +154,293 @@ class CNEMONumerics : public CNumerics {
+                               su2double val_laminar_viscosity,
+                               su2double val_eddy_viscosity,
+                               su2double val_thermal_conductivity,
+                               su2double val_thermal_conductivity_ve,
+                               su2double val_dist_ij, const su2double *val_normal,
+                               su2double val_dS, const su2double *val_Fv,
+                               su2double **val_Jac_i, su2double **val_Jac_j,
+                               const CConfig *config){
+
+    /*--- Initialize matrices ---*/
+    Matrix<NVAR, NVAR> dFdVi, dFdVj, dVdUi, dVdUj;
+    /*--- Need to avoid size of 1 otherwise the type becomes a vector (without [i][j]). ---*/
+    constexpr unsigned short nSpeciesDummy = (NSPECIES == 1) ? 2 : NSPECIES;
+    Matrix<nSpeciesDummy, nSpeciesDummy> dJdr_i, dJdr_j;
+
+    /*--- Play tricks on the compiler, in static mode use NVAR from the template, in dynamic mode
+          use nVar from the class or from the arguments (if you make it static it needs to be an argument). ---*/
+    const auto nVar = NVAR;
+    const auto nSpecies = NSPECIES;
+    const auto nDim = nVar - nSpecies - 2;
+
+    // Allocate and initialize, for the static case the compiler optimizes this away.
+    dFdVi.resize(nVar, nVar) = su2double(0.0);
+    dFdVj.resize(nVar, nVar) = su2double(0.0);
+    dVdUi.resize(nVar, nVar) = su2double(0.0);
+    dVdUj.resize(nVar, nVar) = su2double(0.0);
+    if (NSPECIES == DynamicSize) {
+      dJdr_i.resize(nSpecies,nSpecies) = su2double(0.0);
+      dJdr_j.resize(nSpecies,nSpecies) = su2double(0.0);
+    }
+
+    /*--- Calculate preliminary geometric quantities ---*/
+    su2double dij = val_dist_ij;
+    su2double theta = GeometryToolbox::SquaredNorm(nDim, val_normal);
+
+    /*--- Rename for convenience ---*/
+    su2double rho_i = V_i[RHO_INDEX];
+    su2double rho_j = V_j[RHO_INDEX];
+    su2double T   = val_Mean_PrimVar[T_INDEX];
+    su2double Tve = val_Mean_PrimVar[TVE_INDEX];
+    su2double mu  = val_laminar_viscosity;
+    su2double ktr = val_thermal_conductivity;
+    su2double kve = val_thermal_conductivity_ve;
+    su2double RuSI= UNIVERSAL_GAS_CONSTANT;
+    su2double Ru  = 1000.0*RuSI;
+    const auto& Ds  = val_diffusion_coeff;
+    const auto& hs = fluidmodel->ComputeSpeciesEnthalpy(T, Tve, val_Mean_Eve);
+    const auto& Cvtr = fluidmodel->GetSpeciesCvTraRot();
+    const auto& Ms = fluidmodel->GetSpeciesMolarMass();
+
+    /*--- Calculate useful diffusion parameters ---*/
+    // Mass fraction and summation of diffusion fluxes
+    su2double sumY = 0.0;
+    su2double sumY_i = 0.0;
+    su2double sumY_j = 0.0;
+    for (auto iSpecies = 0; iSpecies < nSpecies; iSpecies++) {
+      Ys[iSpecies]   = val_Mean_PrimVar[RHOS_INDEX+iSpecies];
+      Ys_i[iSpecies] = V_i[RHOS_INDEX+iSpecies]/V_i[RHO_INDEX];
+      Ys_j[iSpecies] = V_j[RHOS_INDEX+iSpecies]/V_j[RHO_INDEX];
```

Comment:
If the `Ys` variables are only used here, you can use the same strategy as with the matrices, simply use `Matrix<NSPECIES, 1>` to get a column vector.